overlord/devicestate/firstboot_test.go

```diff
@@ -58,75 +59,82 @@ import (
 	"github.com/snapcore/snapd/timings"
 )
 
-type FirstBootTestSuite struct {
+type firstBootBaseTest struct {
 	testutil.BaseTest
 
 	systemctl *testutil.MockCmd
 
-	// TestingSeed helps populating seeds (it provides
-	// MakeAssertedSnap, WriteAssertions etc.) for tests.
-	*seedtest.TestingSeed
-
 	devAcct *asserts.Account
 
 	overlord *overlord.Overlord
 
 	perfTimings timings.Measurer
 }
 
-var _ = Suite(&FirstBootTestSuite{})
-
-func (s *FirstBootTestSuite) SetUpTest(c *C) {
-	s.BaseTest.SetUpTest(c)
+func (t *firstBootBaseTest) setupBaseTest(c *C, s *seedtest.SeedSnaps) {
+	t.BaseTest.SetUpTest(c)
 
 	tempdir := c.MkDir()
 	dirs.SetRootDir(tempdir)
-	s.AddCleanup(func() { dirs.SetRootDir("/") })
+	t.AddCleanup(func() { dirs.SetRootDir("/") })
 
-	s.AddCleanup(release.MockOnClassic(false))
+	t.AddCleanup(release.MockOnClassic(false))
 
 	// mock the world!
 	err := os.MkdirAll(filepath.Join(dirs.SnapSeedDir, "snaps"), 0755)
 	c.Assert(err, IsNil)
-	err = os.MkdirAll(filepath.Join(dirs.SnapSeedDir, "assertions"), 0755)
-	c.Assert(err, IsNil)
 
 	err = os.MkdirAll(dirs.SnapServicesDir, 0755)
 	c.Assert(err, IsNil)
 	os.Setenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS", "1")
-	s.AddCleanup(func() { os.Unsetenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS") })
-	s.systemctl = testutil.MockCommand(c, "systemctl", "")
-	s.AddCleanup(s.systemctl.Restore)
-
-	err = ioutil.WriteFile(filepath.Join(dirs.SnapSeedDir, "seed.yaml"), nil, 0644)
-	c.Assert(err, IsNil)
+	t.AddCleanup(func() { os.Unsetenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS") })
+	t.systemctl = testutil.MockCommand(c, "systemctl", "")
+	t.AddCleanup(t.systemctl.Restore)
 
-	s.TestingSeed = &seedtest.TestingSeed{}
-	s.SetupAssertSigning("can0nical", s)
+	s.SetupAssertSigning("canonical")
 	s.Brands.Register("my-brand", brandPrivKey, map[string]interface{}{
 		"verification": "verified",
 	})
 
-	s.SnapsDir = filepath.Join(dirs.SnapSeedDir, "snaps")
-	s.AssertsDir = filepath.Join(dirs.SnapSeedDir, "assertions")
-
-	s.devAcct = assertstest.NewAccount(s.StoreSigning, "developer", map[string]interface{}{
+	t.devAcct = assertstest.NewAccount(s.StoreSigning, "developer", map[string]interface{}{
 		"account-id": "developerid",
 	}, "")
 
-	s.AddCleanup(ifacestate.MockSecurityBackends(nil))
+	t.AddCleanup(sysdb.InjectTrusted([]asserts.Assertion{s.StoreSigning.TrustedKey}))
```

Comment:
so for clarity, we are now doing this here because the other seed* tests don't really need the key to be trusted, but device state does, so we set this up here instead of doing it deep inside SeedSnaps (which then is embedded inside TestingSeed16 which then is embedded inside firstBoot16Suite, which also embeds firstBootBaseTest)

Reply:
yes, also depending on the choice of the account-id of the root of trust, there might be clashes. The original code in seedtest was injecting both an "account" and an "account-key", while here we inject only an "account-key".  

So this relates to the uniformity switch s/"can0nical"/"canonical" for the account-id of the root of trust.